gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/UserServiceImpl.java

```diff
@@ -262,7 +261,14 @@ public UserEntity create(final RegisterUserEntity registerUserEntity) {
 
             // Set date fields
             user.setUpdatedAt(new Date());
-            user = userRepository.update(user);
+
+            if (checkUser.isPresent()) {
```

Comment:
Thanks for this PR and sorry for the delay... 
Actually, here the user is confirming his previous registration so we can consider that is existing yet and we should not allow here to create a user without the email validation process. So the update should be enough here. Could you please make the change?

Reply:
yes sure!